charts/nri-kube-events/templates/_helpers_compatibility.tpl

```diff
@@ -3,8 +3,144 @@ Returns a dictionary with legacy runAsUser config.
 We know that it only has "one line" but it is separated from the rest of the helpers because it is a temporary things
 that we should EOL. The EOL time of this will be marked when we GA the deprecation of Helm v2.
 */}}
-{{- define "newrelic.compatibility.securityContext.pod" -}}
-{{- if  .Values.runAsUser -}}
+{{- define "nri-kube-events.compatibility.securityContext.pod" -}}
+{{- if .Values.runAsUser -}}
 runAsUser: {{ .Values.runAsUser }}
 {{- end -}}
 {{- end -}}
+
+
+
+{{/*
+Creates the image string needed to pull the integration image respecting the breaking change we made in the values file
+*/}}
+{{- define "nri-kube-events.compatibility.images.integration" -}}
+{{- /* workaround: https://github.com/helm/helm/issues/9266 */ -}}
+{{- $values := (.Values | merge (dict)) -}}
+
+{{- $oldRegistry := dig "image" "kubeEvents" "registry" "" $values }}
```

Comment:
is dig available in all helm3 versions?

I looked around and it does not seems so
https://stackoverflow.com/questions/68321891/jupyterhub-helm-install-function-dig-not-defined

Reply:
It was added to sprig in the version [v3.2.0](https://github.com/Masterminds/sprig/releases/tag/v3.2.0): https://github.com/Masterminds/sprig/commit/4bb6f1cc0f550e6ad5198043c72d5da830673a83

They added it to Helm in the version [v3.6.0](https://github.com/helm/helm/releases/tag/v3.6.0): https://github.com/helm/helm/commit/3dbb1614c94c96b34122c8ef788c1c899ff391d4

Helm 3.6.0 is from the 27 of May 2021. I don't think we have reasons to think this could be an issue, right?

Reply:
I would ask Mikel/Roberto, I fear that it could be an issue 😕

Reply:
If I'm not wrong this could cause problems to customers using Helm v3.0.x or v3.1.x
Am I right?

If yes, we should be careful on this, we already asked our customers to migrate to Helm v3, even I guess new customers will update to the latest (3.8.2).

Is there any workaround possible?

Reply:
I'll do a workaround :)

Reply:
Solved in [#9a19c07](https://github.com/newrelic/nri-kube-events/pull/117/commits/9a19c070d4e2903356e9be3eb00fc8e47c103908)